src/compile/mark/mixins.ts

```diff
@@ -314,42 +314,39 @@ export function bandPosition(
 ) {
   const scaleName = model.scaleName(channel);
   const sizeChannel = channel === 'x' ? 'width' : 'height';
+  const {markDef, encoding, config} = model;
 
-  if (model.encoding.size != null || model.markDef.size != null || defaultSizeRef?.value !== undefined) {
-    const orient = model.markDef.orient;
-    if (orient) {
-      const centeredBandPositionMixins = {
-        // Use xc/yc and place the mark at the middle of the band
-        // This way we never have to deal with size's condition for x/y position.
-        [channel + 'c']: ref.fieldRef(fieldDef, scaleName, {}, {band: 0.5})
-      };
+  const vgChannel = alignedChannel(channel, markDef, config);
 
-      if (getTypedFieldDef(model.encoding.size)) {
-        return {
-          ...centeredBandPositionMixins,
-          ...nonPosition('size', model, {vgChannel: sizeChannel})
-        };
-      } else if (isValueDef(model.encoding.size)) {
+  const centeredBandPositionMixins = {
+    [vgChannel]: ref.fieldRef(fieldDef, scaleName, {}, {band: 0.5})
+  };
+
+  if (encoding.size != null || markDef.size != null) {
```

Comment:
Should we use `!=` or `!==` here?

Reply:
`!=` is useful when you also want to support null and undefined.

Reply:
@kanitw said we want to be clearer in terms of semantics with `x !== null && x !== undefined`. Did you guys agree on which one to use?

Reply:
`!= null` is a common pattern and I think it's good to use. In all other cases, we want to be explicit.